src/components/common/Navbar.tsx

```diff
@@ -58,22 +72,71 @@ const Navbar: React.FC = () => {
       <div className="flex items-center gap-x-4">
         <Image src="/assets/icons/moonIcon.svg" width={14} height={17} alt="" />
         <div className="w-[1px] bg-gray-300 h-6"></div>
-        <SignInButton
-          nonce={getNonce}
-          onSuccess={handleSuccess}
-          onError={() => {
-            setError(true)
-          }}
-          onSignOut={() => signOut()}
-        />
+
+        {session ? (
+          <DropdownMenu>
+            <DropdownMenuTrigger>
+              <Button
+                className="gap-x-2.5 text-sm font-medium"
+                variant="outline"
+              >
+                {" "}
+                <Avatar className="!w-6 !h-6">
+                  <AvatarImage src={session?.user?.image || ""} alt="avatar" />
+                  <AvatarFallback>{session.user?.name}</AvatarFallback>
+                </Avatar>{" "}
+                {session.user?.name}
+                <Image
+                  src="/assets/icons/arrowDownIcon.svg"
+                  width={10}
+                  height={6}
+                  alt=""
+                />
+              </Button>
+            </DropdownMenuTrigger>
+            <DropdownMenuContent>
+              <DropdownMenuItem
+                className="cursor-pointer"
+                onClick={() => signOut()}
+              >
+                Logout
+              </DropdownMenuItem>
+            </DropdownMenuContent>
+          </DropdownMenu>
+        ) : (
+          <MemorizedSignInButton
+            nonce={getNonce}
+            expirationTime={handleGetExpiryTime}
```

Comment:
does it make sense to have any sort of expiration checks here if we're going to read from the next-auth context for authenticated users?